Add ethereumjs to list of execution clients

scorbajio opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
This change adds the ethereumjs client to the list of execution clients.

holgerd77 commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -171,6 +172,12 @@ Nethermind is an Ethereum implementation created with the C# .NET tech stack, li
+
+It is written in TypeScript and composed of a number of packages, including core Ethereum primitives represented by the the Block, Transaction, and Merkle-Patricia Trie classes and core client components including an implementation of the Ethereum Virtual Machine (EVM), a blockchain class, and the DevP2P networking stack.
+
+Learn more about ethereumjs by reading its [documentation](https://ethereumjs.readthedocs.io/en/latest/)
```
We initially had a more cautious text here as well, emphasizing the experimental/research focus of the client. I am not sure where this got lost but we should really stick with that, otherwise we are overpromising here as well.

scorbajio replied:
Updated.

holgerd77 reviewed (commented):
This is not yet in a state where this should be merged, we will continue to refine here.

scorbajio commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -133,13 +133,14 @@ The Ethereum community maintains multiple open-source execution clients (previou
+| [Besu](https://besu.hyperledger.org/en/stable/) | Java       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
+| [Erigon](https://github.com/ledgerwatch/erigon) | Go         | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Reth](https://github.com/paradigmxyz/reth)     | Rust       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Ethereumjs](https://github.com/ethereumjs)     | Typescript | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Full                         | Archive, Pruned |
```
Must have been a misunderstanding during internal discussion. Updated  🙂.

holgerd77 commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -171,6 +172,14 @@ Nethermind is an Ethereum implementation created with the C# .NET tech stack, li
 
+### Ethereumjs {#ethereumjs}
+
+It is written in TypeScript and composed of a number of packages, including core Ethereum primitives represented by the the Block, Transaction, and Merkle-Patricia Trie classes and core client components including an implementation of the Ethereum Virtual Machine (EVM), a blockchain class, and the DevP2P networking stack.
```
It -> "The EthereumJS Client"

holgerd77 replied:
Update: or maybe "The EthereumJS Execution Client (EthereumJS)" here, and the "It" for the next paragraph below.

scorbajio replied:
Fixed.

holgerd77 commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -171,6 +172,14 @@ Nethermind is an Ethereum implementation created with the C# .NET tech stack, li
 
 Nethermind also has [detailed documentation](https://docs.nethermind.io), strong dev support, an online community and 24/7 support available for premium users.
 
+### Ethereumjs {#ethereumjs}
```
Can we do "EthereumJS" here?

holgerd77 commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -171,6 +172,14 @@ Nethermind is an Ethereum implementation created with the C# .NET tech stack, li
+
+The EthereumJS Execution Client (EthereumJS) has been in active development for some time now. Although it is not production-ready, it already serves a variety of purposes like testing, research (e.g. EIPs) and developer tooling to engage with the protocol.
+
+Learn more about ethereumjs by reading its [documentation](https://ethereumjs.readthedocs.io/en/latest/).
```
"ethereumjs" -> "EthereumJS"

This  documentation is outdated and should not be referenced, please refer to the monorepo instead.

scorbajio replied:
Fixed. Changed the link to refer to the monorepo, as well.

holgerd77 reviewed (commented):
Some additional commnents.

holgerd77 commented on src/content/developers/docs/nodes-and-clients/index.md:
```diff
@@ -133,13 +133,14 @@ The Ethereum community maintains multiple open-source execution clients (previou
+| [Besu](https://besu.hyperledger.org/en/stable/) | Java       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
+| [Erigon](https://github.com/ledgerwatch/erigon) | Go         | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Reth](https://github.com/paradigmxyz/reth)     | Rust       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Ethereumjs](https://github.com/ethereumjs)     | Typescript | Linux, Windows, macOS | Sepolia, Goerli, and more          | Full                               | Archive         |
```
Ah, here also: "Ethereumjs" -> "EthereumJS"

Also: "Typescript" -> "TypeScript"

holgerd77 replied:
The URL here should point to the client package, or the monorepo at most (but client package is better)

wackerow reviewed (commented):
Hey @scorbajio apologies that this PR was missed... So we've since updated this list ([/developers/docs/nodes-and-clients/](https://ethereum.org/en/developers/docs/nodes-and-clients/)) to include EthereumJS, but I see slight differences in what was published and your PR...

Should we remove "Mainnet" from the EthereumJS listing? We have this listed, but also have EthereumJS labeled as beta to act as a warning against using this for production.

Your PR only notes "Full" sync... we currently have listed "Snap, Full, Light"... should we remove snap/full?

We've updated these to all show "Mainnet, Sepolia, Holesky"... We've removed Goerli as it's being deprecated.... can we confirm that EthereumJS is compatible with Holesky?

If you or anyone else is able to reply here, I'm happy to make any updates within your PR and bring them in.

Happy to make any updates needed

scorbajio commented:
> Hey @scorbajio apologies that this PR was missed... So we've since updated this list ([/developers/docs/nodes-and-clients/](https://ethereum.org/en/developers/docs/nodes-and-clients/)) to include EthereumJS, but I see slight differences in what was published and your PR...
> 
> Should we remove "Mainnet" from the EthereumJS listing? We have this listed, but also have EthereumJS labeled as beta to act as a warning against using this for production.
> 
> Your PR only notes "Full" sync... we currently have listed "Snap, Full, Light"... should we remove snap/full?
> 
> We've updated these to all show "Mainnet, Sepolia, Holesky"... We've removed Goerli as it's being deprecated.... can we confirm that EthereumJS is compatible with Holesky?
> 
> If you or anyone else is able to reply here, I'm happy to make any updates within your PR and bring them in.
> 
> Happy to make any updates needed

Hey, thanks for the update!

We like to stay on the cautious side (promise less than what we deliver), so let's exclude `Mainnet` from the list of networks and `Snap` as well as `Light` from the list of sync strategies. There is also the short paragraph description from this same PR which I see wasn't included in the updates, which could help provide some more details on the EthereumJS client:
```
### EthereumJS {#ethereumjs}

The EthereumJS Execution Client (EthereumJS) is written in TypeScript and composed of a number of packages, including core Ethereum primitives represented by the the Block, Transaction, and Merkle-Patricia Trie classes and core client components including an implementation of the Ethereum Virtual Machine (EVM), a blockchain class, and the DevP2P networking stack.

Learn more about it by reading its [documentation](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master)
```

wackerow reviewed (approved):
Great, thanks @scorbajio! Updated per your suggestions. Added a commit that nests the description under an "In development" header to make sure this remains clean. We've tended to hold off on adding these sections until out of beta, but we really just need to make sure it's obvious to readers. Pulling in!
